File: src/main/java/x/samantha/commandsjda/util/CommandSearchUtil.java
package x.samantha.commandsjda.util;

import org.apache.lucene.analysis.standard.StandardAnalyzer;
import org.apache.lucene.document.Document;
import org.apache.lucene.document.Field;
import org.apache.lucene.document.StringField;
import org.apache.lucene.index.IndexWriter;
import org.apache.lucene.index.IndexWriterConfig;
import org.apache.lucene.store.RAMDirectory;
import x.samantha.commandsjda.managers.CommandManager;

import java.io.IOException;

@SuppressWarnings("deprecation")
public class CommandSearchUtil {
    public static StandardAnalyzer analyzer;
    public static RAMDirectory index;

    CommandSearchUtil() {
        analyzer = new StandardAnalyzer();
        index = new RAMDirectory();
        IndexWriterConfig cfg = new IndexWriterConfig(analyzer);
        IndexWriter writer;
        try {
            writer = new IndexWriter(index, cfg);
            CommandManager.getCommandRegistry().keySet().forEach(key -> {
                Document doc = new Document();
                doc.add(new StringField("command", key, Field.Store.YES));
                try {
                    writer.addDocument(doc);
                } catch (IOException ignored) {}
            });
            writer.close();
        } catch (IOException ignored) {}
    }
}

File: src/main/kotlin/x/samantha/commandsjda/util/CommandSearchUtil.kt
package x.samantha.commandsjda.util

import x.samantha.commandsjda.managers.CommandManager
import org.apache.lucene.analysis.standard.StandardAnalyzer
import org.apache.lucene.document.Document
import org.apache.lucene.document.Field
import org.apache.lucene.document.StringField
import org.apache.lucene.index.IndexWriter
import org.apache.lucene.index.IndexWriterConfig
import org.apache.lucene.store.RAMDirectory

class CommandSearchUtil {

    companion object {
        lateinit var analyzer: StandardAnalyzer
        lateinit var index: RAMDirectory
    }

    init {
        analyzer = StandardAnalyzer()
        index = RAMDirectory()
        val config = IndexWriterConfig(analyzer)
        val writer = IndexWriter(index, config)

        CommandManager.commandRegistry.keys.forEach {
            val doc = Document()
            doc.add(StringField("command", it, Field.Store.YES))
            writer.addDocument(doc)
        }

        writer.close()
    }
}